pkg/ccl/changefeedccl/changefeedbase/settings.go

```diff
@@ -158,3 +158,19 @@ var EventMemoryMultiplier = settings.RegisterFloatSetting(
 		return nil
 	},
 )
+
+// ProtectTimestampInterval controls the frequency of protected timestamp record updates
+var ProtectTimestampInterval = settings.RegisterDurationSetting(
+	"changefeed.protect_timestamp_interval",
+	"controls how often the changefeed forwards its protected timestamp to the resolved timestamp",
+	10*time.Minute,
+	settings.PositiveDuration,
+)
+
+// ActiveProtectedTimestamps enables always having protected timestamps laid
+// down that are periodically advanced to the highwater mark.
+var ActiveProtectedTimestamps = settings.RegisterBoolSetting(
+	"changefeed.active_protected_timestamps",
+	"if set, rather than only protecting changefeed targets from garbage collection during backfills, data will always be protected up to the changefeed's frontier",
+	false,
```

Comment:
I think we need to add this flag into retiredSettings in registry.go (on master).